logstash-core/src/main/java/org/logstash/Event.java

```diff
@@ -209,15 +211,12 @@ public String toJson() throws JsonProcessingException {
         Object o = JSON_MAPPER.readValue(json, Object.class);
         // we currently only support Map or Array json objects
         if (o instanceof Map) {
-            result = new Event[]{ new Event((Map)o) };
+            result = new Event[]{ new Event((Map<String, Object>)o) };
         } else if (o instanceof List) {
-            result = new Event[((List) o).size()];
+            result = new Event[((Collection<?>) o).size()];
             int i = 0;
-            for (Object e : (List)o) {
-                if (!(e instanceof Map)) {
-                    throw new IOException("incompatible inner json array object type=" + e.getClass().getName() + " , only hash map is supported");
-                }
-                result[i++] = new Event((Map)e);
+            for (Map<String, Object> e : (Iterable<Map<String, Object>>) o) {
```

Comment:
While correct, the code is a bit hard to read. We check if it is List, then upcast to a Collection , the iterate over it as a Iterable.  Can we just cast to List<?> once and use that variable through out ?

Reply:
@jakelandis yea def! Will fix this soon :)